### DISCUSSION TITLE
Queue improvement. Any tag. Header stick at top.

### DISCUSSION THREAD

**Discussion by Emilot** [jcorporation/myMPD#423] [Ideas]
Let me start:

1) Bring "Any Tag" everywhere (implemented already)

2) Make the Header stick only at Desktop resolutions

3) Let's improve Playlists JC. My thought is that Smart playlist are useless. Some feedback from Collybia OS says that when some people have Smart Playlists "ON", produce hundreds useless playlists, that overflow their "Playlist" card view.

I suggest:

1a) At "Queue" card when i want to Save a new Playlist (my queue) i should have a dropdown menu with my already saved Playlists (right now there is only a field for enter the name of the playlist) to choose one and "Add" my present queue songs or to make a new one.
2a) At Queue card i want to choose the settings "Replay", "Shuffle"(already there) from a dropdown menu.

**Comment by jcorporation** [comment]
3) smart playlists

What you mean is not the smart playlist feature itself. It is the smart playlist generation feature, that can create smart playlists based on tags. Default is one playlist per genre tag. This behavior can be disabled in the settings. I have accurately tagged music files and I love this feature.

**Reply by Emilot** [reply]
I didn't write it right. Smart Playlists are ok, because someone can disable them. Nothing else you can do and it is ok. The magic word here is "accurate". I guess a lot of people do not have the "perfect" organized library and the feature "Smart Playlists" is a mess for them and only for them. Nothing else about it.

My propose was for improve Playlists. Not Smart Playlists. My mistake.

**Comment by jcorporation** [comment]
2) sticky header

only the toolbar or the table header itself also?

**Reply by Emilot** [reply]
Only the toolbar!

**Comment by jcorporation** [comment]
1a) Save queue or add to playlist 
Save queue is a native feature of mpd. Your suggestion that the queue can be added to an already existing playlist is not implemented in mpd. That must be manually coded in the myMPD backend. Evtl. you can propose such a feature for mpd itself.

**Reply by Emilot** [reply]
Point taken! I will suggest it to Max :)

**Comment by jcorporation** [comment]
2a queue controls

is on the roadmap #414

**Reply by Emilot** [reply]
🥇

**Comment by jcorporation** [comment]
>     2\. Make the Header stick only at Desktop resolutions

I am played around a little bit to implement your feature request and implemented a sticky table header. I see now that you requested a sticky toolbar... but I like the experience with the sticky table header. It is configurable in the advanced settings dialog. It is work in progress and eventually ships with the next version (default is old behaviour).

**Reply by Emilot** [reply]
No problem for either of two implementations JC! I will take a look but i am sure i will like it whatever you decide!

**Reply by jcorporation** [reply]
I reverted the commit. Sticky table headers collidates with responsive tables from bootstrap.

**Reply by Emilot** [reply]
Too bad... Well, thank you for the try JC!
